### ISSUE TITLE
Syndicating WP content to Tumblr includes sharedaddy div

### ISSUE THREAD

**Issue by chrisaldrich** [Automattic/jetpack#4399]
#### Steps to reproduce the issue
I'm using [Social Network Auto-Poster](https://wordpress.org/plugins/social-networks-auto-poster-facebook-twitter-g/) (rather than Jetpack's Publicize because it includes more control over post types and additional platforms) to syndicate my WP content to my Tumblr. I'm using their standard "Text Post" option with the text set to include `<p><img src='%IMG%'/></p><p>%FULLTEXT%</p>`

#### What I expected
The copy on Tumblr should simply transfer only the user's input in the post from WP into the Tumblr post.
 
#### What happened instead
I get the content of the post, but I'm also getting a div with all the class=sharedaddy etc. (see below) appended to the bottom of the post which generates the Like prompt/button/functionality and the word "Loading...". (Example: See bottom of http://chrisaldrich.tumblr.com/post/142661877457/a-new-view-of-the-tree-of-life which was syndicated from http://boffosocko.com/2016/04/11/a-new-view-of-the-tree-of-life/)

Here's the HTML that's being copied over and displaying unnecessarily.
`<div class="sharedaddy sd-block sd-like jetpack-likes-widget-wrapper jetpack-likes-widget-unloaded" id="like-post-wrapper-67433065-0-570c56a68fcae" data-src="//widgets.wp.com/likes/#blog_id=67433065&amp;post_id=0&amp;origin=boffosocko.com&amp;obj_id=67433065-0-570c56a68fcae" data-name="like-post-frame-67433065-0-570c56a68fcae"><h3 class="sd-title">Like this:</h3><div class="likes-widget-placeholder post-likes-widget-placeholder" style="height:55px"><span class="button"><span>Like</span></span> <span class="loading">Loading...</span></div><span class="sd-text-color"></span><a class="sd-link-color"></a></div></div>
`

Generally, this doesn't happen when syndicating content like this into other platforms (though this could be because other platforms, Facebook for example, may be stripping out all other html and thus precluding this). Additionally, for example, it's not copying over Jetpack's "Share This" or "Related" posts content which also sit inside the main `<div class="e-content">` which I suspect is the general section being copied over. 

In short: Why is the "Like" functionality being copied over while "Sharing" and "Related" aren't?

**Comment by jeherve** [comment]
That's interesting. Could you let me know what your sharing settings are, under Settings > Sharing in your dashboard?

**Comment by chrisaldrich** [comment]
<a href="https://github.com/Automattic/jetpack/issues/4399" class="u-in-reply-to"></a>I've got all the services enabled, "icons only", sharing label: "Share this:"; all four options on "Show buttons on" are checked; likes are "turned on per post"; "Disable CSS and JS" is unchecked. 

As an additional test (for possible recreation and comparison/contrast) I'm also using a plugin called Syndication links [https://wordpress.org/plugins/syndication-links/] which generally hooks in between Jetpack's "Share This" and "Related Posts". I'm using the default display "Syndicated to:" which is followed by several social icons. When I syndicate to Tumblr, similar to the "like" functionality from Jetpack described above, I also often have the text "Syndicated to:" displayed on the post, (but without the icons as I believe they're SVG's which reside somewhere else).

(As an aside, functionality like Syndication Links would be a cool feature for Jetpack to add/acquire, particularly if the "Publicize" functionality could hook into it and automatically pull in the links where the content was shared.)

Thanks @jeherve!

**Comment by jeherve** [comment]
Thanks for the extra details. I believe this may be linked with how both Likes and Syndication links are hooked into `the_content`, WordPress' filter to output the post content:
https://codex.wordpress.org/Plugin_API/Filter_Reference/the_content

For both Likes and Sharing buttons, we go through a series of checks to figure out if the buttons should be added to `the_content`:
- https://github.com/Automattic/jetpack/blob/4.1.1/modules/sharedaddy/sharing-service.php#L586
- https://github.com/Automattic/jetpack/blob/4.1.1/modules/likes.php#L1016

Unfortunately, it looks like we're not catching SNAP's situation well enough for Likes.

 Could you try to apply the following patch to `modules/likes.php` on your site, and let me know if it helps?
https://github.com/jeherve/jetpack/commit/4ca6c985033a9d6d2e45d15864bb0726f8877d16

Thanks!

**Comment by dshanske** [comment]
I developed the Syndication Links plugin to fill a need for me. Rel-syndication indicates a syndicated copy of a post on a site, as opposed to a traditional share link which allows the site visitor to share the Post on that site. If there is anything I can do to help  @chrisaldrich on my end, please advise.

**Comment by chrisaldrich** [comment]
<a href="https://github.com/Automattic/jetpack/issues/4399#issuecomment-232799361" class="u-in-reply-to"></a><a href="http://stream.boffosocko.com/2016/syndicating-wp-content-to-tumblr-includes-sharedaddy-div-issue-4399" class="u-in-reply-to"></a><a href="http://stream.boffosocko.com/2016/ive-got-all-the-services-enabled-icons-only-sharing-label" class="u-in-reply-to"></a><a href="https://github.com/Automattic/jetpack/issues/4399" class="u-in-reply-to"></a>@jeherve, It would appear that that patch seems to take care of the issue.  I ran a few quick tests on some variations on the settings and didn't see the offending sharedaddy div appear after the patch was applied. 
Compare: http://boffosocko.com/2016/07/19/the-memory-palace-of-matteo-ricci/ and http://chrisaldrich.tumblr.com/post/147683457752/the-memory-palace-of-matteo-ricci

I'll test a few additional things over the next few days (I recall another social platform acting somewhat like Tumblr had, but don't recall which at the moment), but otherwise this seems to fix things.

@dshanske Perhaps these same types of checks (or the patch code mentioned) could serve as a template to prevent the similar issue within Syndication Links which shows "Syndicated to" on several platforms including Tumblr and Facebook. I'll open an issue there as a reminder.

**Comment by jeherve** [comment]
> It would appear that that patch seems to take care of the issue.

Excellent! Let me know how things turn out in the few days; if everything works well, I'll see about adding that patch to Jetpack.

**Comment by chrisaldrich** [comment]
<a href="https://github.com/Automattic/jetpack/issues/4399#issuecomment-233285986" class="u-in-reply-to"></a>That moment, when you think everything is fixed and then... http://chrisaldrich.tumblr.com/post/147983482257/introduction-to-complex-analysis-ucla-extension 

The only thing I think I've changed since what I thought was previously fine, was changing a default in settings from having "likes" on per post to having "likes" on for all posts as a default.  To my knowledge I haven't changed any other settings. Simultaneously, I'm not sure what/how this would/could shift the change... thoughts?

**Comment by jeherve** [comment]
@chrisaldrich Is the patch still applied on your Jetpack install?

**Comment by chrisaldrich** [comment]
Sorry, I should have mentioned it explicitly, but I did doubleckeck and the patch was still there. (I'd put it in manually and couldn't remember having installed any Jetpack patches lately, but checked just in case it may have been overwritten.)

**Comment by jeherve** [comment]
@chrisaldrich Sorry to hear that :(

Could you try to move my patch a bit downer, like so, and let me know how it goes?
https://github.com/Automattic/jetpack/commit/027e815b0e1c308544c62db203c4376a69f77d22

Thanks!

**Comment by jeherve** [comment]
Closing this. Don't hesitate to reopen if this is still an issue for you!
